src/main/java/teammates/ui/webapi/ResetAccountAction.java

```diff
@@ -20,17 +20,19 @@ public JsonResult execute() {
         }
 
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-        String wrongGoogleId = null;
+
         if (studentEmail != null) {
             StudentAttributes existingStudent = logic.getStudentForEmail(courseId, studentEmail);
             if (existingStudent == null) {
                 throw new EntityNotFoundException("Student does not exist.");
             }
-            wrongGoogleId = existingStudent.getGoogleId();
+
+            String existingGoogleId = existingStudent.getGoogleId();
 
             try {
-                logic.resetStudentGoogleId(studentEmail, courseId);
-                taskQueuer.scheduleCourseRegistrationInviteToStudent(courseId, studentEmail, true);
+                if (isAccountMigrated(existingGoogleId)) {
+                    sqlLogic.resetStudentGoogleId(studentEmail, courseId, existingGoogleId);
+                }
```

Comment:
What if the account has not been migrated? This needs to work with both DBs. We need to maintain the original functionality as well, as was done in your previous PRs.

Reply:
Sorry, I interpreted wrongly from your previous review. Will make the changes!